dot/peerset/peerset_test.go

```diff
@@ -25,96 +86,98 @@ func TestPeerSetBanned(t *testing.T) {
 
 	// we ban a node by setting its reputation under the threshold.
 	rpc := newReputationChange(BannedThresholdValue-1, "")
+
+	mock.ExpectByCall(Message{Status: Drop, setID: 0x0, PeerID: "testPeer1"})
 	// we need one for the message to be processed.
 	handler.ReportPeer(rpc, peer1)
 	time.Sleep(time.Millisecond * 100)
 
-	checkMessageStatus(t, <-ps.resultMsgCh, Drop)
-
+	mock.ExpectByCall(Message{Status: Reject, setID: 0x0, PeerID: "testPeer1"})
 	// check that an incoming connection from that node gets refused.
 	handler.Incoming(0, peer1)
-	checkMessageStatus(t, <-ps.resultMsgCh, Reject)
 
 	// wait a bit for the node's reputation to go above the threshold.
 	time.Sleep(time.Millisecond * 1200)
 
+	mock.ExpectByCall(Message{Status: Accept, setID: 0x0, PeerID: "testPeer1"})
 	// try again. This time the node should be accepted.
 	handler.Incoming(0, peer1)
-	require.NoError(t, err)
-	checkMessageStatus(t, <-ps.resultMsgCh, Accept)
+
+	mock.Assert(t)
```

Comment:
Instead of making all assertions together, it would be nicer to make assertions after each respective function.

Also, since assertions are happening in mockProcessMessage.Assert method, it does not give us a clear picture of what is being tested. I think it would be better if you can move those assertions in individual tests.

This applies to other tests in this file as well.

You could bring back `checkMessageStatus` in individual tests if possible.